### PR TITLE
Map modification using at working.

### PR DESCRIPTION
**Pull request by OndrejSpanel**
I have implemented "at" for Maps by copying and modifying "at" for sequences.

Basic tests provided and passed, no current tests broken.

### CODE REVIEW

**Comment by OndrejSpanel** [comment]
I have also added each for completeness. It performs mapValues and will mostly likely be not used much, but it look clean and simple, and provides nice symmetry with traversables.

This should complete the issue #15.

**Comment by OndrejSpanel** [comment]
Do you have some estimation when this could be merged and published? It is kind of blocking for me in my project and I am not aware of any sensible workaround.

File: tests/src/test/scala/com/softwaremill/quicklens/ModifyMapAtTest.scala [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.softwaremill.quicklens
+
+import com.softwaremill.quicklens.TestData._
+import org.scalatest.{ FlatSpec, ShouldMatchers }
+
+class ModifyMapAtTest extends FlatSpec with ShouldMatchers {
+
+  it should "modify a non-nested map with case class item" in {
+    modify(m1)(_.at("K1").a5.name).using(duplicate) should be(m1dup)
+    //modify(l1)(_.at(2)).using(a3 => modify(a3)(_.a4.a5.name).using(duplicate)) should be(l1at2dup)
```
Comment (adamw) [inline tests/src/test/scala/com/softwaremill/quicklens/ModifyMapAtTest.scala]: remove this?

File: tests/src/test/scala/com/softwaremill/quicklens/TestData.scala [hunk 2]
```diff
@@ -57,4 +57,12 @@ object TestData {
   val ll1at2at1dup = List(List(A5("d1"), A5("d2")), List(A5("d3"), A5("d4"), A5("d5")), List(A5("d6"), A5("d7d7")))
   val ll1at2eachdup = List(List(A5("d1"), A5("d2")), List(A5("d3"), A5("d4"), A5("d5")), List(A5("d6d6"), A5("d7d7")))
   val ll1eachat1dup = List(List(A5("d1"), A5("d2d2")), List(A5("d3"), A5("d4d4"), A5("d5")), List(A5("d6"), A5("d7d7")))
+
+  case class M2(m3: Map[String, A4])
+
+  val m1 = Map("K1" -> A4(A5("d1")), "K2" -> A4(A5("d2")), "K2" -> A4(A5("d3")))
```
Comment (adamw) [inline tests/src/test/scala/com/softwaremill/quicklens/TestData.scala]: shouldn't the third key be `K3`?

**Comment by adamw** [comment]
Ah! one more thing - probably for a separate PR - updating the readme docs for `.at` :)

**Comment by adamw** [comment]
1.4.5 released, should be in central soon

**Comment by OndrejSpanel** [comment]
Great, thanks a lot.

It was a pleasure for me to work with this - the library looks cleanly designed, and your extra quick PR handling deserves kudos.

**Comment by adamw** [comment]
Thank you for the contributions :)

**Pull request merged**